corehq/apps/users/decorators.py

```diff
@@ -79,3 +79,13 @@ def _inner(request, domain, couch_user_id, *args, **kwargs):
         else:
             return domain_specific_login_redirect(request, domain)
     return _inner
+
+
+def ensure_active_user_by_username(username):
+    ccu = CommCareUser.get_by_username(username)
+    valid, message, error_code = True, None, None
+    if ccu and not ccu.is_active:
+        valid, message, error_code = False, 'User deactivated', 'mobile.app.translation.user.is.deactivated'
+    elif CommCareUser.get_deleted_by_username(username):
+        valid, message, error_code = False, 'User deleted', 'mobile.app.translation.user.is.deleted'
```

Comment:
Looks good! How about `user.deactivated.on.server` and `user.deleted.on.server` for translations keys?
And `'User has been deactivated, please reactivate on server to submit data'` and `'User has been deleted on server, you will not be able to submit collected data to server'` for default messages.
`406` response code sounds good